bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/service/TivoStatusProvider.java

```diff
@@ -0,0 +1,491 @@
+                }
+                this.tivoSocket = tivoSocket;
+            } else {
+                logger.debug(" connSocketConnect '{}' - socket creation failed to host '{}', port '{}'",
+                        tivoConfigData.getCfgIdentifier(), tivoConfigData.getCfgHost(), tivoConfigData.getCfgTcpPort());
+                return false;
+            }
+
+            return true;
+
+        } catch (UnknownHostException e) {
+            logger.debug(" TiVo '{}' - while connecting, unexpected host error: '{}'",
+                    tivoConfigData.getCfgIdentifier(), e.getMessage());
+        } catch (IOException e) {
+            if (tivoStatusData.getConnectionStatus() != ConnectionStatus.OFFLINE) {
+                logger.debug(" TiVo '{}' - I/O exception while connecting: '{}'", tivoConfigData.getCfgIdentifier(),
+                        e.getMessage());
+            }
+        }
+        return false;
+    }
+
+    /**
+     * {@link doNappTime} sleeps for the period specified by the getCfgCmdWait parameter. Primarily used to allow the
+     * TiVo time to process responses after a command is issued.
+     */
+    public void doNappTime() {
+        try {
+            logger.debug(" doNappTime '{}' - I feel like napping for '{}' milliseconds",
+                    tivoConfigData.getCfgIdentifier(), tivoConfigData.getCfgCmdWait());
+            TimeUnit.MILLISECONDS.sleep(tivoConfigData.getCfgCmdWait());
+        } catch (Exception e) {
+        }
+    }
+
+    public TivoStatusData getServiceStatus() {
+        return tivoStatusData;
+    }
+
+    public void setServiceStatus(TivoStatusData tivoStatusData) {
+        this.tivoStatusData = tivoStatusData;
+    }
+
+    /**
+     * {@link StreamReader} data stream reader that reads the status data returned from the TiVo.
+     *
+     */
+    public class StreamReader extends Thread {
+        private @Nullable BufferedReader bufferedReader = null;
+        private volatile boolean stopReader;
+
+        private CountDownLatch stopLatch;
+
+        /**
+         * {@link StreamReader} construct a data stream reader that reads the status data returned from the TiVo via a
+         * BufferedReader.
+         *
+         * @param inputStream socket input stream.
+         * @throws IOException
+         */
+        public StreamReader(InputStream inputStream) {
+            this.setName("OH-binding-" + tivoConfigData.getCfgIdentifier() + "-" + tivoConfigData.getCfgHost() + "."
+                    + tivoConfigData.getCfgTcpPort());
+            this.bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+            this.stopLatch = new CountDownLatch(1);
+            this.setDaemon(true);
+        }
+
+        @Override
+        public void run() {
+            try {
+                logger.debug("streamReader {} is running. ", tivoConfigData.getCfgIdentifier());
+                while (!stopReader && !Thread.currentThread().isInterrupted()) {
```

Comment:
You could invoke `Thread.interrupt()` if you want to stop the thread. That would make the flag unnecessary.

Reply:
I removed the flag